tools/check_documentation.py

```diff
@@ -0,0 +1,170 @@
+
+from praw.util.cache import cachedproperty  # noqa: E402
+from praw.models.reddit.base import RedditBase  # noqa: E402
+from praw.models.reddit.modmail import ModmailObject  # noqa: E402
+
+
+class DocumentationChecker:
+    """Checks for code block statements and attribute tables in subclasses.
+
+    Attribute EXCEPTIONS holds all exceptions. All classes and subclasses in
+    the exceptions list will be ignored.
+
+    Attribute EXPLICIT_INCLUDE holds all classes that should be included
+    regardless of what is in the exceptions.
+
+    Attribute METHOD_EXCEPTIONS holds the names of methods that will be
+    filtered out.
+    """
+
+    HAS_CODE_BLOCK = re.compile(r".. code-block::")
+    HAS_ATTRIBUTE_TABLE = re.compile(r"Attribute[ ]+Description")
+
+    BASE_SEARCH_CLASS = RedditBase
+    SUBCLASSES = []
+    EXCEPTIONS = [
+        ModmailObject,  # is never publicly accessed
+    ]
+    EXPLICIT_INCLUDE = []
+
+    METHOD_EXCEPTIONS = [
+        "parse",
+        "sluggify",
+        "id_from_url",
+        "from_data",
+    ]
+
+    @staticmethod
+    def in_list_lambda(search_list):
+        return lambda item: item not in search_list
+
+    @classmethod
+    def expand_subclasses(cls):
+        """Recursively find subclasses"""
+        base_list = list(cls.BASE_SEARCH_CLASS.__subclasses__())
+        while len(base_list) > 0:
+            item = base_list[0]
+            base_list = base_list[1:]
+            cls.SUBCLASSES.append(item)
+            base_list.extend(item.__subclasses__())
+
+    @classmethod
+    def expand_exceptions(cls):
+        """Recursively find exceptions"""
+        base_list = cls.EXCEPTIONS
+        cls.EXCEPTIONS = []
+        while len(base_list) > 0:
+            item = base_list[0]
+            base_list = base_list[1:]
+            cls.EXCEPTIONS.append(item)
+            base_list.extend(item.__subclasses__())
+
+    @classmethod
+    def check(cls):
+        cls.expand_subclasses()
+        cls.expand_exceptions()
+        success = True  # Initially, there is success
+        for subclass in set(
+            list(
+                filter(cls.in_list_lambda(cls.EXCEPTIONS), cls.SUBCLASSES)
+            )  # Filters out all exceptions from the subclasses
+            + cls.EXPLICIT_INCLUDE
+            # adds the classes that should always be in the list
+        ):  # set conversion to exclude all duplicates
+            doc = subclass.__doc__  # extracts the documentation
+            if doc is None:  # Proceeds if there is no documentation
+                print(
+                    "Subclass {mod}.{name} is missing documentation.".format(
+                        mod=subclass.__module__, name=subclass.__name__
+                    )
+                )
+                success = False
+            else:  # Proceeds if there is documentation
+                has_attribute_table = bool(cls.HAS_ATTRIBUTE_TABLE.search(doc))
```

Comment:
No need for `bool`, as you can directly do the conditional, and explicitly set `success = False` in the block.